### ISSUE TITLE
Deep link gives ERR_UNKNOWN_URL_SCHEME by default

### ISSUE THREAD

**Issue by The-First-King** [Slion/Fulguris#598]
# Description
It's not possible to add new account in DAVx5 when using Fulguris web-browser due to the following error shown below.

```
Web page not available

The web page at at.bitfire.davdroid:/oauth2/redirect?state=abc&code=abc&scope=https://www.googleapis.com/auth/calendar%20https://www.googleapis.com/auth/carddav could not be loaded because:

net::ERR_UNKNOWN_URL_SCHEME
```

NOTE: Last time it might be working fine with Fulguris v1.9.13 (not sure). But the Android System WebView version was different. It has been tested in the middle of October 2023.

# Environment
 - Device: any
 - OS: LineageOS
 - Version: 19.1

# To reproduce
Steps to reproduce the behaviour:
1. Open DAVx5 and click on the “+” button.
2. Select the radio bar with Google Contacts / Calendar and click the “LOGIN” button.
3. In the Google account text field type the e-mail address and click enter button on the keyboard to proceed with redirection.  Fulguris web-browser will be opened. 
4. After confirming Google account’s credential the error message will appear.

# Expected behaviour
Redirection back to the DAVx5 application should work after Google authentication completes in Fulguris.

# Workaround
The problem is not present when changing the default web-browser in Android OS from Fulguris to Jelly (default web-browser in LineageOS).

**Comment by Slion** [comment]
I don't think anything has been changed in Fulguris in this regard lately. Also I believe the error you are getting comes from the WebView itself. Though I'm confident this could be worked around in Fulguris somehow. Unlikely to look at it any time soon however.

**Comment by The-First-King** [comment]
Yea, it may be related to the WebView version. Currently, I'm using the latest version 120.0.6099.210. The default version is disabled, but it is 112.0.5615.48.

When I used a workaround, I disabled both Fulguris and WebView v120.0.6099.210. It works with Jelly and WebView v112.0.5615.48 i.e. how it would be if I wouldn't install Fulguris and the latest WebView version.

**Comment by The-First-King** [comment]
It appears that the Dropbox plugin for Ghost Commander is also impacted by the same problem. Good news - it doesn't related to the WebView implementation. Fulguris v1.9.8 is working fine. It looks to be related to some regression in the web-browser with permission level requests.

**Comment by The-First-King** [comment]
I have found an exact release when the problem appears. It is Fulguris v1.9.17. The last working version is 1.9.16.

Fulguris doesn’t ask for permission to open a third party application, and this behavior is causing the “net::ERR_UNKNOWN_URL_SCHEME” error message.

**Comment by Slion** [comment]
Nice investigation. Try change the Launch app option default domain settings to Ask or Yes and see if that helps.

**Comment by The-First-King** [comment]
Indeed, it works well now! Just tested with the latest Fulguris v1.9.24.

**Comment by Slion** [comment]
Ho dear, I should have thought of that earlier really. I've had such deep links issues myself lately too. I just did not connect the dots. Launch app option should be set to Yes by default as that's what most people need and expect.

Did you try with both Yes and Ask? Did it work as expected in both cases?

Again, good job on that investigation and thanks for supporting Fulguris.

**Comment by Slion** [comment]
Related to #548, #542.

**Comment by The-First-King** [comment]
Redirection to 3rd party apps are no longer working after the fix. Now it doesn't matter what is set Yes or Ask in default domain settings.

**Comment by Slion** [comment]
That's not cool. What do you get then?

**Comment by The-First-King** [comment]
API request authorization is appearing:

![01](https://github.com/Slion/Fulguris/assets/59452275/273b9f1b-34b9-490a-a01c-b95f8c1f3d65)

When tapping Allow button, nothing happens. No errors, no redirection, or anything,

![02](https://github.com/Slion/Fulguris/assets/59452275/5b72ef1b-a42c-44c7-8c07-99c36e0223fc)

Unfortunately, Fulguris is not usable in this scenario.
